File: backend/src/domain/entities/AcademicRecord.ts
import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm';

@Entity('academic_records')
export class AcademicRecord {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'member_id' })
  memberId: number;

  @Column({ type: 'varchar', name: 'student_code', nullable: true })
  studentCode: string;

  @Column({ type: 'varchar', name: 'degree_type', nullable: true })
  degreeType: string;

  @Column({ type: 'varchar', name: 'class_name', nullable: true })
  className: string;

  @Column({ type: 'integer', name: 'start_year', nullable: true })
  startYear: number;

  @Column({ type: 'integer', name: 'graduated_year', nullable: true })
  graduatedYear: number;

  @Column({ type: 'varchar', nullable: true })
  status: string;
}

File: backend_java/src/main/java/com/service/backend/domain/entity/AcademicRecord.java
package com.service.backend.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("academic_records")
public class AcademicRecord {

    @Id
    private Long id;

    @Column("member_id")
    private Long memberId;

    @Column("student_code")
    private String studentCode;

    @Column("degree_type")
    private String degreeType;

    @Column("class_name")
    private String className;

    @Column("start_year")
    private Integer startYear;

    @Column("graduated_year")
    private Integer graduatedYear;

    private String status;
}
